src/models/ResourceType.ts

```diff
@@ -43,10 +86,18 @@ export class ResourceType extends Typegoose {
   // endregion
 
   // region public methods
+  @instanceMethod
+  public updateFromObject(updateObject: any) {
+    ResourceTypeModel.schema.eachPath( (path) => {
```

Comment:
Arrow function is a duplicate to `ResourceInstance.ts`, extract that code into an own function.